js/binance.js

```diff
@@ -1873,39 +1930,49 @@ module.exports = class binance extends Exchange {
     parseMarket (market) {
         let swap = false;
         let future = false;
+        let option = false;
+        const underlying = this.safeString (market, 'underlying');
         const id = this.safeString (market, 'symbol');
+        const optionParts = id.split ('-');
+        const optionBase = this.safeString (optionParts, 0);
         const lowercaseId = this.safeStringLower (market, 'symbol');
-        const baseId = this.safeString (market, 'baseAsset');
+        const baseId = this.safeString (market, 'baseAsset', optionBase);
         const quoteId = this.safeString (market, 'quoteAsset');
-        const settleId = this.safeString (market, 'marginAsset');
+        const settleId = this.safeString (market, 'marginAsset', 'USDT');
         const base = this.safeCurrencyCode (baseId);
         const quote = this.safeCurrencyCode (quoteId);
         const settle = this.safeCurrencyCode (settleId);
         const contractType = this.safeString (market, 'contractType');
-        const contract = ('contractType' in market);
-        const spot = !contract;
-        let expiry = this.safeInteger (market, 'deliveryDate');
+        let contract = ('contractType' in market);
+        let expiry = this.safeInteger2 (market, 'deliveryDate', 'expiryDate');
         if ((contractType === 'PERPETUAL') || (expiry === 4133404800000)) { // some swap markets do not have contract type, eg: BTCST
             expiry = undefined;
             swap = true;
+        } else if (underlying !== undefined) {
+            contract = true;
+            option = true;
         } else {
             future = true;
         }
+        const spot = !contract;
         const filters = this.safeValue (market, 'filters', []);
         const filtersByType = this.indexBy (filters, 'filterType');
         const status = this.safeString2 (market, 'status', 'contractStatus');
         let contractSize = undefined;
         let fees = this.fees;
         let linear = undefined;
         let inverse = undefined;
+        const strike = this.safeString (market, 'strikePrice');
         let symbol = base + '/' + quote;
         if (contract) {
             if (swap) {
                 symbol = symbol + ':' + settle;
             } else if (future) {
                 symbol = symbol + ':' + settle + '-' + this.yymmdd (expiry);
+            } else if (option) {
+                symbol = symbol + ':' + settle + '-' + this.yymmdd (expiry) + '-' + strike + '-' + this.safeString (optionParts, 3);
```

Comment:
@Dan-krm binance send the values with full precision but I think we should probably parse them as ints otherwise the unified symbol will be weird
'ETH/USDT:USDT-230224-800.00000000-C` instead of `'ETH/USDT:USDT-230224-800-C`

![image](https://user-images.githubusercontent.com/43336371/217816223-b145831f-12eb-4f7f-a014-bec77468212b.png)

Reply:
Yeah that makes sense, that does look pretty weird and none of the strike prices use any decimal places currently